src/main/java/net/minecraftforge/client/extensions/IForgeEntityModel.java

```diff
@@ -0,0 +1,31 @@
+package net.minecraftforge.client.extensions;
+
+import net.minecraft.client.model.geom.ModelPart;
+import net.minecraft.world.entity.Entity;
+import net.minecraftforge.client.model.animation.IEntityModelAnimationModifier;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public interface IForgeEntityModel<T extends Entity>
+{
+    Set<IEntityModelAnimationModifier> animationModifiers = new HashSet<>();
+
+    default void prepareAnim(T p_102618_, float p_102619_, float p_102620_, float p_102621_, float p_102622_, float p_102623_)
+    {
+        setupAnim(p_102618_, p_102619_, p_102620_, p_102621_, p_102622_, p_102623_);
+        ModelPart root = getModelRoot();
+
+        for (IEntityModelAnimationModifier modifier : animationModifiers)
+            modifier.modifyAnimations(p_102618_, root, p_102619_, p_102620_, p_102621_, p_102622_, p_102623_);
+    }
+
+    void setupAnim(T p_102618_, float p_102619_, float p_102620_, float p_102621_, float p_102622_, float p_102623_);
```

Comment:
This method being named such that it is implemented by the same-named MC methods in `EntityModel` may be a bit iffy on the reobfuscation. My thinking is that, because of reobfuscation of MC methods, the corresponding method in `HumanoidModel` will be reobfuscated back into a SRG name and therefore will not be implementing the method defined by this interface, causing a crash when the JVM verifies the class.

If you are able, please test this in a userdev environment by making a mod with your PR and running it in production. Tse `publishToMavenLocal` to publish the necessary artifacts needed by userdev and then use the version indicated in the Gradle console output. Use `installerJar` to build the installer JAR so you can test it in production.